packml_ros/src/packml_ros.cpp

```diff
@@ -180,43 +196,58 @@ void PackmlRos::getCurrentStats(packml_msgs::Stats& out_stats)
 {
   packml_sm::PackmlStatsSnapshot stats_snapshot;
   sm_->getCurrentStatSnapshot(stats_snapshot);
+  out_stats = populateStatsMsg(stats_snapshot);
+}
+
+
+void PackmlRos::getStatsTransaction(packml_msgs::Stats &out_stats)
+{
+  packml_sm::PackmlStatsSnapshot stats_snapshot;
+  sm_->getCurrentStatTransaction(stats_snapshot);
+  out_stats = populateStatsMsg(stats_snapshot);
+}
 
-  out_stats.idle_duration.data.fromSec(stats_snapshot.idle_duration);
-  out_stats.exe_duration.data.fromSec(stats_snapshot.exe_duration);
-  out_stats.held_duration.data.fromSec(stats_snapshot.held_duration);
-  out_stats.susp_duration.data.fromSec(stats_snapshot.susp_duration);
-  out_stats.cmplt_duration.data.fromSec(stats_snapshot.cmplt_duration);
-  out_stats.stop_duration.data.fromSec(stats_snapshot.stop_duration);
-  out_stats.abort_duration.data.fromSec(stats_snapshot.abort_duration);
-  out_stats.duration.data.fromSec(stats_snapshot.duration);
-  out_stats.fail_count = stats_snapshot.fail_count;
-  out_stats.success_count = stats_snapshot.success_count;
-  out_stats.availability = stats_snapshot.availability;
-  out_stats.performance = stats_snapshot.performance;
-  out_stats.quality = stats_snapshot.quality;
-  out_stats.overall_equipment_effectiveness = stats_snapshot.overall_equipment_effectiveness;
-
-  out_stats.error_items.clear();
+packml_msgs::Stats PackmlRos::populateStatsMsg(const packml_sm::PackmlStatsSnapshot& stats_snapshot)
+{
+  packml_msgs::Stats stats_msg;
+
+  stats_msg.idle_duration.data.fromSec(stats_snapshot.idle_duration);
+  stats_msg.exe_duration.data.fromSec(stats_snapshot.exe_duration);
+  stats_msg.held_duration.data.fromSec(stats_snapshot.held_duration);
+  stats_msg.susp_duration.data.fromSec(stats_snapshot.susp_duration);
+  stats_msg.cmplt_duration.data.fromSec(stats_snapshot.cmplt_duration);
+  stats_msg.stop_duration.data.fromSec(stats_snapshot.stop_duration);
+  stats_msg.abort_duration.data.fromSec(stats_snapshot.abort_duration);
+  stats_msg.duration.data.fromSec(stats_snapshot.duration);
+  stats_msg.fail_count = stats_snapshot.fail_count;
+  stats_msg.success_count = stats_snapshot.success_count;
+  stats_msg.availability = stats_snapshot.availability;
+  stats_msg.performance = stats_snapshot.performance;
+  stats_msg.quality = stats_snapshot.quality;
+  stats_msg.overall_equipment_effectiveness = stats_snapshot.overall_equipment_effectiveness;
+
+  stats_msg.error_items.clear();
```

Comment:
Why does the error_items field need to be cleared? The message is instantiated in the current scope.

Reply:
Good call, removed.